12-offer-encoding.md

```diff
@@ -0,0 +1,1057 @@
+    2. data:
+        * [`...*u64`:`incoming_msat`]
+    1. type: 20 (`issuer`)
+    2. data:
+        * [`...*utf8`:`issuer`]
+    1. type: 30 (`node_id`)
+    2. data:
+        * [`point32`:`node_id`]
+    1. type: 32 (`quantity`)
+    2. data:
+        * [`tu64`:`quantity`]
+    1. type: 34 (`refund_for`)
+    2. data:
+        * [`sha256`:`refunded_payment_hash`]
+    1. type: 36 (`recurrence_counter`)
+    2. data:
+       * [`tu32`:`counter`]
+    1. type: 54 (`send_invoice`)
+    1. type: 68 (`recurrence_start`)
+    2. data:
+        * [`tu32`:`period_offset`]
+    1. type: 64 (`recurrence_basetime`)
+    2. data:
+        * [`tu64`:`basetime`]
+    1. type: 38 (`payer_key`)
+    2. data:
+        * [`point32`:`key`]
+    1. type: 39 (`payer_note`)
+    2. data:
+        * [`...*utf8`:`note`]
+    1. type: 50 (`payer_info`)
+    2. data:
+        * [`...*byte`:`blob`]
+    1. type: 40 (`created_at`)
+    2. data:
+        * [`tu64`:`timestamp`]
+    1. type: 42 (`payment_hash`)
+    2. data:
+        * [`sha256`:`payment_hash`]
+    1. type: 44 (`relative_expiry`)
+    2. data:
+        * [`tu32`:`seconds_from_creation`]
+    1. type: 46 (`cltv`)
+    2. data:
+        * [`tu32`:`min_final_cltv_expiry`]
+    1. type: 48 (`fallbacks`)
+    2. data:
+        * [`byte`:`num`]
+        * [`num*fallback_address`:`fallbacks`]
+    1. type: 52 (`refund_signature`)
+    2. data:
+        * [`bip340sig`:`payer_signature`]
+    1. type: 56 (`replace_invoice`)
+    2. data:
+        * [`sha256`:`payment_hash`]
+    1. type: 240 (`signature`)
+    2. data:
+        * [`bip340sig`:`sig`]
+
+1. subtype: `blinded_payinfo`
+2. data:
+   * [`u32`:`fee_base_msat`]
+   * [`u32`:`fee_proportional_millionths`]
+   * [`u16`:`cltv_expiry_delta`]
+   * [`u16`:`flen`]
+   * [`flen*byte`:`features`]
+
+1. subtype: `fallback_address`
+2. data:
+   * [`byte`:`version`]
+   * [`u16`:`len`]
+   * [`len*byte`:`address`]
+
+## Requirements
+
+A writer of an invoice:
+  - MUST set `created_at` to the number of seconds since Midnight 1
+    January 1970, UTC when the offer was created.
+  - MUST set `payment_hash` to the SHA2 256-bit hash of the
+    `payment_preimage` that will be given in return for payment.
+  - MUST set (or not set) `send_invoice` the same as the offer.
+  - MUST set `offer_id` to the id of the offer.
+  - MUST specify exactly one signature TLV: `signature`.
+    - MUST set `sig` to the signature using `node_id` as described in [Signature Calculation](#signature-calculation).
+  - if the chain for the invoice is not solely bitcoin:
+    - MUST specify `chains` the invoice is valid for.
+  - otherwise:
+    - the bitcoin chain is implied as the first and only entry.
+  - if it has bolt11 features:
+    - MUST set `features` to the bitmap of features.
+  - if the invoice corresponds to an offer with `recurrence`:
+    - MUST set `recurrence_basetime` to the start of period #0 as calculated
+      by [Period Calculation](#offer-period-calculation).
```

Comment:
How? I assume as a number of seconds since 01/01/1970 but it should be explicit.